pandas/core/ops.py

```diff
@@ -444,18 +409,82 @@ def _validate(self, lvalues, rvalues, name):
-              self.is_datetime_rhs):
+        else:
+            raise TypeError('cannot operate on a series without a rhs '
+                            'of a series/ndarray of type datetime64[ns] '
+                            'or a timedelta')
+
+    def _validate_timedelta(self, name):
+        # assumes self.is_timedelta_lhs
+
+        if self.is_integer_rhs or self.is_floating_rhs:
+            # timedelta and integer mul/div
+            self._check_timedelta_with_numeric(name)
+        elif self.is_timedelta_rhs or self.is_offset_rhs:
+            # 2 timedeltas
+            if name not in ('__div__', '__rdiv__', '__truediv__',
+                            '__rtruediv__', '__add__', '__radd__', '__sub__',
+                            '__rsub__'):
+                raise TypeError("can only operate on a timedeltas for addition"
+                                ", subtraction, and division, but the operator"
+                                " [{name}] was passed".format(name=name))
+        elif self.is_datetime_rhs:
+            if name not in ('__add__', '__radd__', '__rsub__'):
+                raise TypeError("can only operate on a timedelta/DateOffset "
+                                "with a rhs of a datetime for addition, "
+                                "but the operator [{name}] was passed"
+                                .format(name=name))
+        else:
+            raise TypeError('cannot operate on a series without a rhs '
+                            'of a series/ndarray of type datetime64[ns] '
+                            'or a timedelta')
+
+    def _validate_offset(self, name):
+        # assumes self.is_offset_lhs
+
+        if self.is_timedelta_rhs:
+            # 2 timedeltas
+            if name not in ('__div__', '__rdiv__', '__truediv__',
+                            '__rtruediv__', '__add__', '__radd__', '__sub__',
+                            '__rsub__'):
+                raise TypeError("can only operate on a timedeltas for addition"
+                                ", subtraction, and division, but the operator"
+                                " [{name}] was passed".format(name=name))
 
+        elif self.is_datetime_rhs:
             if name not in ('__add__', '__radd__'):
                 raise TypeError("can only operate on a timedelta/DateOffset "
                                 "and a datetime for addition, but the operator"
                                 " [{name}] was passed".format(name=name))
+
         else:
             raise TypeError('cannot operate on a series without a rhs '
                             'of a series/ndarray of type datetime64[ns] '
                             'or a timedelta')
 
+    def _validate(self, lvalues, rvalues, name):
+        if self.is_datetime_lhs:
+            return self._validate_datetime(lvalues, rvalues, name)
+        elif self.is_timedelta_lhs:
+            return self._validate_timedelta(name)
+        elif self.is_offset_lhs:
```

Comment:
I guess this is ok for now. Maybe should make 3 sub-classes here (or I think ultimately refactoring to have *all* this logic in the index methods as it already duplicates some of this).